### ISSUE TITLE
GGMAX Bug - Batch Import  triggers intermittent thumbnail adjustment screens.  (40-50% of the time)

### ISSUE THREAD

**Issue by casigus** [TheGameCreators/GameGuruRepo#5758]
I doubt its a bug, but its very annoying.

NOW! the concept of the batch importer is 10/10. love the concept.

but when your importing 330 assets from blender, and you have to click a button for each one being imported, it kinda takes away the batch element of it, a batch importer should automaticly import the model to the libary with out having to be constantly pressing add every few seconds. the repition of that is honnestly making me want to stick my head in a hole and cry into my sock ya know?

any solutions or fixes for this would be amazinggggggggg. 

or an option to batch import and it automates from there and taking away the constant clicking, the constant clicking on my chamber door, 

As of some one gently clicking, clicking at my chamber door.
“’Tis some model format,” I muttered, “clicking at my chamber door—
            Only this and nothing more.”

Quoth the Raven “click me more.”

    “Prophet!” said I, “thing of evil!—prophet still, if bird or devil! Or FBX or Gltf!!!!”

Quoth the Raven “click me more.”

k thanks. bye.

**Comment by Kasseyus** [comment]
Feature/Enhancement Request to remove the thumbnail adjustment element of the batch model import process.

**Comment by Kasseyus** [comment]
@LeeBamberTGC 

When performing a batch import of assets, the thumbnail generator will trigger intermittently, so for every 10 objects batch imported, it may pop-up 4-5 times in that process,  but yet all of  the  objects will  import successfully.

Given the nature of batch importing, it may be better to simply remove the requirement to generate a thumbnail on batch imports.

**Comment by casigus** [comment]
110% ive been building up a libary of asssets and im ready to import, but theres 100s. did  the first lot, it took 4 days. kept the pc on and had to manaually click add,  there in now, but it was a very very tedious process. is there anything I can do to change it on my end? the packs are ready but im not sitting throught hat again.

**Comment by Kasseyus** [comment]
@casigus No, I'm  afraid not.  The thumbnail generator screen is  triggered when importing an asset to allow adjustment of the thumbnail, add keywords, background, etc.    It's like a final confirmation screen  and was introduced  before  the 'Batch Import' command existed.

**Comment by casigus** [comment]
yeah makes sense. gonna hold off importing my packs till this can be sorted. but yeah its not a batch importer if its keeps popping up, on top of that, I have to select the import folder for the batch as the same folder as where I want the models going, so if my models are on a another drive after exporting from blender, i'd then have to copy alllllll those models into the files/entitymap/foldername directory which is also counter productive! 

ideally you should be able to batch import from any folder into your desired location.

IF you already have.dbos in the batch folder, it will also re-import them which again, not great. can also change some custom edits which is very annoying!

thanks! :)

**Comment by Kasseyus** [comment]
@casigus What I do when importing batch objects, is to copy the FBX/DDS files of the models I want to convert into their own folder, and move that into the user directory and import that folder.

Then I can delete the residual FBX files and then just move the folder to where it needs to go.  This way, it's sorted and named correctly with minimal fuss - plus the original files are safe in their own development directory just in case I need to perform a re-import.

**Comment by casigus** [comment]
yah i  do that but it still pops off every few mins asking you to add to the libary which goes against the point of it being a batch importer. used a .dll viewer to have a nosey at the assimp.dll.  possibly this specific part. 

![{5707C484-4A41-454C-B0D8-B3A3DD1FC3BC}](https://github.com/user-attachments/assets/dc96c6ba-1dce-42cd-a334-723ecd0c5792)

howwver its beyond my capability to do anything with that info!

**Comment by Kasseyus** [comment]
@LeeBamberTGC 

When performing a batch import of assets, the thumbnail generator will trigger intermittently, so for every 10 objects batch imported, it may pop-up a couple of times in that process, but yet all of the objects will import successfully and thumbnails will still be generated.

Given the nature of batch importing, the thumbnail screen should not be called at all, and thumbnails generated in the background without user input.

**Comment by casigus** [comment]
thats exactly it yeah!!  I had to install a auto clicker because it was driving me madddddddddddd.  (THE AUTO clicker doesn't work as the import button shifts around !)

**Comment by LeeBamberTGC** [comment]
@casigus I have added functionality that will skip preview thumb adjustment window when using the batch converter. I have also added a tick box in the batch converter UI so that you can skip any DBOs found in the import folder, so you are only targetting all the non-DBO models you have in there :)  Let me know if it works for you, currently in the DEVTEST build and will be in the EXP at some point after community testing, and in the next public update in June 2025.

**Comment by casigus** [comment]
To Lee and TheGameCreators team,
I appreciate the recent updates to the batch converter in GameGuru MAX. It's good to see some steps forward, even if small ones.
However, I need to be honest about my experience.
Over the last five years, I have invested thousands of pounds into GameGuru MAX assets through both your DLCs and the Game Creator Store. Despite every effort to make the engine work — from optimizing textures, converting formats, editing game files manually, and using LOD tools — GameGuru MAX remains fundamentally unstable for any serious project. Crashes, bugs, and poor performance continue to make it impossible to complete anything of substance.
I was promised one-on-one support, which never materialized. Reaching out to your community often left me feeling ignored or dismissed, even before any frustration set in. Over time, this not only cost me time and money, but also the entirety of a project I had been building for years — a devastating loss.
Adding to the difficulty, GameGuru MAX’s proprietary .dbo format has locked all the assets I invested in. I cannot use them in any other engine, and no conversion method has been provided.
At this stage, I have two reasonable requests:
🔹 Either provide a tool or method to allow the conversion of my assets into a usable format (such as FBX) so I can move my work into Unreal Engine,
🔹 Or refund the money spent on GameGuru MAX DLCs and Creator Store assets.
I have been advised by Steam’s legal department and my bank, and have given them a full report outlining the issues. However, I would prefer a resolution directly through you if possible. I believe it is only fair, given the investment made in good faith.
I am fully aware of TheGameCreators' company structure, size, and filings. I also recognize that with a team of 12, it is unlikely GameGuru MAX will ever reach the level needed for serious game development. That’s not a personal criticism — it’s simply reality given the scope of modern engines.
I am asking for understanding and a practical solution — not confrontation. I simply want a way to preserve my work or to be fairly reimbursed for what has been lost.
Thank you for your time.

________________________________

**Comment by LeeBamberTGC** [comment]
@casigus Zip up and send me all your DBO model files and texture files, and I will see what I can do.  Can I ask how you produced the DBO files in the first instance?

**Comment by casigus** [comment]
theres  582 GB of files. textures. models.

so this would be very difficult.

all the .dbos where produced from importing them into ggmax.

ive been developing a tool to convert .dbos into fbxs,  so far progress is very very very very slow, every attempt has just spat out jumbled models.

[cid:9aac92c1-4652-46d4-9598-f93204998e72]

Source
Amount
Steam DLC Purchases
£723.82 GBP
Game Creator Store Purchases
£1,023.02 GBP
TOTAL SPENT
£1,746.84 GBP

ou have spent approximately £1,746.84 GBP on GameGuru MAX assets across Steam and the Creator Store.
✅ Nearly £1,750 worth of investment is now locked in .dbo format and unusable outside of GameGuru MAX.
✅ This includes DLCs, asset packs, sound packs, HUDs, gift kits, creator store purchases, etc.

as a disabled person living on support, £1,746.84 GBP  is a HUGE amount.

Your investment (£1,746.84) would represent about 2.5% of their entire reported assets.
That’s massive for a single customer.
You're not "just another user" — you’re someone who materially contributed to their financial standing.

TheGameCreators Ltd (as of 31 March 2024)
£68,966


so considoring all that, I need a way to save those assets, get them turned into something I can use in unreal and we'll shake hands and I'll leave you be.




________________________________

**Comment by casigus** [comment]
Your defense of the .dbo format is exactly what I expected: hollow, outdated, and insulting.

Locking user-created and user-imported assets into a dead, proprietary format has nothing to do with protecting users. It has everything to do with controlling their work and trapping them inside your failing ecosystem. You don’t encrypt to help creators — you encrypt to make sure they can’t leave.

The .dbo format is a relic from a bygone era: slow, unsupported, and incompatible with every professional tool used today. Every asset converted to .dbo is another asset effectively destroyed. You cripple users’ futures and then have the audacity to pretend it’s for their protection.

Telling users to "keep the original FBX files" is not a solution — it’s an admission of how destructive your system is. You knew exactly what you were doing when you designed it: holding users hostage, bleeding them financially, and ensuring they could never recover their work.

You didn’t just lose me as a user — you confirmed everything wrong with your product, your culture, and your treatment of your community:

You build walls where others build bridges.

You manufacture prisons where others empower dreams.

You insult the intelligence of the very people who supported you.

I have fully documented:

The technical failures of the .dbo system.

The mistreatment of users raising legitimate concerns.

The repeated, suspicious changes of TheGameCreators' office addresses and directors.

Abusive and manipulative behavior by official community figures.

I will never recommend GameGuru MAX to anyone — only warn them.

I’m moving forward with engines that treat creators like professionals, not criminals.
Good luck surviving the future you’ve built. You’re going to need it.

And as for you, Sync:

You are the worst thing that ever happened to the GameGuru MAX community. Your arrogance, your abuse of power, and your contempt for the very people who kept this project alive have been fully documented.

The way you talk down to customers — people who poured thousands into supporting TheGameCreators — is disgraceful. Long before I finally snapped, you were already rude, controlling, and openly hostile to anyone who questioned the broken systems you helped defend.

I have no idea if you’re officially on the payroll, though I have the official staff information — twelve employees recorded. I also have the full history of TheGameCreators’ suspicious office and director changes over the years.

With TheGameCreators, it’s not personal. I just want my assets.
With you, it is very personal.

Every snarky comment, every time you deflected users with real concerns, every time you abused your position, every manipulative tactic you used — documented.

You don't just represent the toxicity that ruined the community — you are the blueprint for it.

[synchromesh62](https://github.com/synchromesh62):

Consistently defends GameGuru MAX, even when serious technical flaws are raised.

Redirects critical users away from public spaces (Discord invite links = move private).

Minimizes and deflects legitimate criticism rather than engaging with it properly.

Gatekeeps the community: controlling the tone, pushing back against  "negativity". - another words if anyone says anything bad about game guru max. 

Fails to acknowledge the seriousness of complaints from paying customers. (prime example, see above)


This is my final communication to you, Sync.

If I receive any further contact from you — direct or indirect — I will immediately file a malicious communications and harassment report with the police.
This is not a threat; it is a formal warning.
You are now fully on notice.

Do not contact me again. 


![Image](https://github.com/user-attachments/assets/912ef35e-60e2-4030-a631-6357fac98f81)

![Image](https://github.com/user-attachments/assets/af5a383d-a598-48f6-a54b-864892388261)


![Image](https://github.com/user-attachments/assets/1f30be14-ecf8-4653-865f-455c9e4265ee)


![Image](https://github.com/user-attachments/assets/846493ac-c3df-4ff2-86c2-9412fea4d691)

**Comment by synchromesh62** [comment]
Locked due to personal information being displayed

**Comment by LeeBamberTGC** [comment]
Hi Casigus,

We appreciate the amount of content you have accumulated over the years and would like to be able to help you with your mission to move to another game-making tool.

For all the DBO model files and textures you have created yourself, we should be able to convert them for you into a format that can be imported into a standard industry format that can be imported into, say, Unity or Unreal.  We would need your assurance (by email) that you created these assets yourself and that you have a legal right to use them in other tools.

Where the DBO model files have been purchased from third parties, you will need to contact those artists to get their permission to use the models in other tools. Most of the artists who sell through TGC's Steam Store often heavily discount their content in the knowledge that the models cannot be used outside of the game maker they are associated with (this ensures their model sales are not cannibalised elsewhere).  In all cases related to models purchased from the "Game Creator Store", you would need to refer to the license agreed with them. TGC does not own or run this store, and permissions must be sought from either the owner of that store or the individual artists (contact info is usually in the purchased download package and the store owner may be able to help track down contact details for them otherwise).

Your work over the years has been amazing and hugely creative, and we wish you the very best in your future career as a game developer and look forward to seeing your vision realised and released into the world for all to enjoy!

Regards,
Lee, TGC.

P.S. You can email me directly at lee@thegamecreators.com when you have assembled your DBO files zip, and I will see what I can do.
